src/partition/repart.c

```diff
@@ -3984,6 +3992,150 @@ static int context_mangle_partitions(Context *context) {
         return 0;
 }
 
+static int split_name_printf(Partition *p) {
+        assert(p);
+
+        const Specifier table[] = {
+                { 't', specifier_string, gpt_partition_type_uuid_to_string(p->type_uuid) },
```

Comment:
i figure this should be gpt_partition_type_uuid_to_string_harder(), i.e. we don't want this to fail for unknown partition types